### PR TITLE
test: doctest for addNullEmbedFilters

### PR DESCRIPTION
**Pull request by steve-chavez**
Towards fixing #2846

### CODE REVIEW

File: src/PostgREST/Plan.hs [hunk 1]
```diff
@@ -552,24 +556,92 @@ addRelatedOrders (Node rp@ReadPlan{order,from} forest) = do
...
--- It fails if operators other than is.null or not.is.null are used.
+-- | Searches for null filters on embeds, e.g. filter for `projects` on `GET /clients?select=*,projects(*)&projects=not.is.null`
+--
+-- Setup:
+--
+-- >>> let nullOp    = OpExpr True (Is TriNull)
+-- >>> let nonNullOp = OpExpr False (Is TriNull)
+-- >>> let notEqOp   = OpExpr True (Op OpNotEqual "val")
+-- >>> :{
+-- -- this resembles the `projects(*)` part on `/clients?select=*,projects(*)`
```
Comment (laurenceisla) [inline src/PostgREST/Plan.hs:567]:
```suggestion
-- -- this represents the `projects(*)` part on `/clients?select=*,projects(*)`
```

File: src/PostgREST/Plan.hs [hunk 2]
```diff
@@ -552,24 +556,92 @@ addRelatedOrders (Node rp@ReadPlan{order,from} forest) = do
...
+--         relJoinConds = [],
+--         relAlias = Nothing, relAggAlias = "clients_projects_1", relHint = Nothing, relJoinType = Nothing, relIsSpread = False, depth = 1
+--       },
+--       subForest = []
+--     }
+--   ]
+-- :}
+--
+-- >>> :{
+-- -- this will resemble the full URL `/clients?select=*,projects(*)&projects=not.is.null`
```
Comment (laurenceisla) [inline src/PostgREST/Plan.hs:587]:
```suggestion
-- -- this represents the full URL `/clients?select=*,projects(*)&projects=not.is.null`
```

File: src/PostgREST/Plan.hs [hunk 3]
```diff
@@ -552,24 +556,92 @@ addRelatedOrders (Node rp@ReadPlan{order,from} forest) = do
...
+--         )
+--       ],
+--       order = [], range_ = fullRange, relName = "clients", relToParent = Nothing, relJoinConds = [], relAlias = Nothing, relAggAlias = "", relHint = Nothing,
+--       relJoinType = Nothing, relIsSpread = False, depth = 0
+--     },
+--     subForest = subForst
+--   }
+-- :}
+--
+-- Don't do anything to the filter if there's no embedding (a subtree) on projects. Assume it's a normal filter.
```
Comment (laurenceisla) [inline src/PostgREST/Plan.hs:610]:
```suggestion
-- It doesn't do anything to the filter if there's no embedding (a subtree) on projects. Assume it's a normal filter.
```

File: src/PostgREST/Plan.hs [hunk 4]
```diff
@@ -552,24 +556,92 @@ addRelatedOrders (Node rp@ReadPlan{order,from} forest) = do
...
+--
+-- >>> ReadPlan.where_ . rootLabel <$> addNullEmbedFilters (readPlanTree nullOp [])
+-- Right [CoercibleStmnt (CoercibleFilter {field = CoercibleField {cfName = "projects", cfJsonPath = [], cfIRType = "", cfTransform = Nothing, cfDefault = Nothing}, opExpr = OpExpr True (Is TriNull)})]
+--
+-- If there's an embedding on projects, then change the filter to use the internal aggregate name (`clients_projects_1`) so the filter can succeed later.
```
Comment (laurenceisla) [inline src/PostgREST/Plan.hs:615]:
```suggestion
-- If there's an embedding on projects, then it changes the filter to use the internal aggregate name (`clients_projects_1`) so the filter can succeed later.
```

File: src/PostgREST/Plan.hs (hunk 3 above)
Comment (steve-chavez) [reply src/PostgREST/Plan.hs:610]: Hm, the imperative word is shorter. I think we should keep it as it is.

Comment (steve-chavez) [reply src/PostgREST/Plan.hs:610]: The hemingway app also always recommends the "active voice".

File: src/PostgREST/Plan.hs (hunk 4 above)
Comment (steve-chavez) [reply src/PostgREST/Plan.hs:615]: Ditto as above

**Review by laurenceisla — APPROVED** [review]

**Pull request merged**